### PR TITLE
Async Labels for chips & picker

### PR DESCRIPTION
**Pull request by amrutha-rajiv**
Chips & picker have the ability to use an async getLabels function to populate display values 

##### **What did you change?**

- changed items list in chips to use Observables
- picker calls getLabels

##### **Reviewers**
* @jgodi 

##### **Checklist (completed via merger)**
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [contributing guide](https://github.com/bullhorn/novo-elements/blob/master/CONTRIBUTING.md)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Visually tested in supported browsers and devices

### CODE REVIEW

File: src/elements/chips/Chips.js [hunk 1]
```diff
@@ -109,21 +112,43 @@ export class NovoChipsElement extends OutsideClick {
...
+                        label: value
+                    });
                 }
-                return { value: v, label: v };
-            });
-        } else {
-            this.items = [];
+            }
+            if (noLabels.length > 0 && this.source && this.source.getLabels && typeof this.source.getLabels === 'function') {
+                console.log('calling getLabels for', noLabels);//eslint-disable-line
```
Comment (jgodi) [inline src/elements/chips/Chips.js]: console log

File: src/elements/chips/Chips.js [hunk 2]
```diff
@@ -109,21 +112,43 @@ export class NovoChipsElement extends OutsideClick {
...
+                this.source.getLabels(noLabels).then(result => {
+                    this.items = this.items.concat(result);
+                    this._items.next(this.items);
```
Comment (jgodi) [inline src/elements/chips/Chips.js]: Need this, with the one below?

File: src/elements/form/FormValidators.js [hunk 3]
```diff
@@ -46,4 +46,21 @@ export class FormValidators {
         }
         return null;
     }
+
+    static custom(control) {
+        console.log('custom!!!');//eslint-disable-line
```
Comment (jgodi) [inline src/elements/form/FormValidators.js]: console

File: src/elements/form/FormValidators.js [hunk 4]
```diff
@@ -46,4 +46,21 @@ export class FormValidators {
...
+        return new Promise((resolve, reject) => {
+            let customValidator = control.customValidator;
+            debugger;
+            if (customValidator && control.value &&
+            (((customValidator.hasOwnProperty('reject') && customValidator.hasOwnProperty('resolve')) || Object.getPrototypeOf(customValidator).hasOwnProperty('then')) ||
+            typeof customValidator === 'function')) {
+                console.log('calling function');//eslint-disable-line
+                customValidator.then(resolve, reject);
+            } else {
+                console.log('just resolving');//eslint-disable-line
```
Comment (jgodi) [inline src/elements/form/FormValidators.js]: same

File: src/elements/form/FormValidators.js [hunk 5]
```diff
@@ -46,4 +46,21 @@ export class FormValidators {
...
+
+    static custom(control) {
+        console.log('custom!!!');//eslint-disable-line
+        return new Promise((resolve, reject) => {
+            let customValidator = control.customValidator;
+            debugger;
+            if (customValidator && control.value &&
+            (((customValidator.hasOwnProperty('reject') && customValidator.hasOwnProperty('resolve')) || Object.getPrototypeOf(customValidator).hasOwnProperty('then')) ||
+            typeof customValidator === 'function')) {
+                console.log('calling function');//eslint-disable-line
```
Comment (jgodi) [inline src/elements/form/FormValidators.js]: same

File: src/elements/toast/ToastService.js [hunk 6]
```diff
@@ -40,7 +40,7 @@ export class NovoToastService {
                 console.error('No parent view container specified for the ToastService. Set it inside your main application. \nthis.toastService.parentViewContainer = view (ViewContainerRef)'); // eslint-disable-line
                 return;
             }
-
+            console.log('toast alert was called');//eslint-disable-line
```
Comment (jgodi) [inline src/elements/toast/ToastService.js]: console

File: src/elements/toast/ToastService.js [hunk 7]
```diff
@@ -62,10 +62,12 @@ export class NovoToastService {
                 REF.destroy();
             }
         }, 300);
+        console.log('hiding toast', this.references);//eslint-disable-line
```
Comment (jgodi) [inline src/elements/toast/ToastService.js]: ditto

File: src/elements/toast/ToastService.js [hunk 8]
```diff
@@ -62,10 +62,12 @@ export class NovoToastService {
...
     }
 
     handleAlert(toast, options) {
         this.setToastOnSession(toast, options);
+        console.log('toast: handleAlert was called');//eslint-disable-line
```
Comment (jgodi) [inline src/elements/toast/ToastService.js]: ditto

File: src/elements/toast/ToastService.js [hunk 9]
```diff
@@ -99,6 +101,7 @@ export class NovoToastService {
         function addClass() {
             toast.animate = true;
         }
+        console.log('showing toast');//eslint-disable-line
```
Comment (jgodi) [inline src/elements/toast/ToastService.js]: ditto

File: src/elements/form/FormValidators.js (hunk 3 above)
Comment (amrutha-rajiv) [inline src/elements/form/FormValidators.js]: i need to undo these changes

File: src/elements/chips/Chips.js (hunk 2 above)
Comment (amrutha-rajiv) [inline src/elements/chips/Chips.js]: @jgodi  yes - it's inside the promise
